sklearnex/dispatcher.py

```diff
@@ -313,7 +317,61 @@ def get_patch_names():
     return list(get_patch_map().keys())
 
 
-def patch_sklearn(name=None, verbose=True, global_patch=False, preview=False):
+def check_entity_loaded(
+    name: Optional[str] = None, modules: Optional[Dict[str, Any]] = None
+) -> Optional[str]:
+    """
+    This function checks if a specified module or class is already loaded in sys.modules.
+
+    Parameters:
+    name (str, optional): The name of the module or class to check. If no name is provided, the function checks for 'sklearn'.
+
+    Returns:
+    str: A warning message if the specified module or class is already loaded, indicating that patch_sklearn()
+    should be called before any import statements from that module or class. If the module or class is not loaded,
+    the function returns None.
+
+    Example:
+    >>> check_entity_loaded()
+    'sklearn or some parts of it are already loaded. patch_sklearn() only affects classes imported *after* calling it.
+    To retrieve patched entities, make sure to call patch_sklearn() before any import statements from sklearn.'
+
+    >>> check_entity_loaded(name='LogisticRegression')
+    'LogisticRegression or some parts of it are already loaded. patch_sklearn() only affects classes imported *after*
+    calling it. To retrieve patched entities, make sure to call patch_sklearn() before any import statements from LogisticRegression.'
+    """
+
+    def _get_loaded_classes():
+        loaded_classes = []
+        for key, module in sys.modules.items():
+            if "sklearn" in key and isinstance(module, types.ModuleType):
+                loaded_classes.extend(
+                    [
+                        cls.__name__
+                        for cls in vars(module).values()
+                        if isinstance(cls, type)
+                    ]
+                )
+        return loaded_classes
+
+    # list of all loaded modules, uses sys.modules per default
+    modules = modules if modules is not None else sys.modules
+
+    # is `name` or anything from sklearn already loaded?
+    loaded = name is None and "sklearn" in modules.keys()
```

Comment:
From checking locally it seems that `from sklearnex import patch_sklearn` will yield a true result from `"sklearn" in modules.keys()` (even if nothing is directly imported from sklearn) - therefore the warning is raised in all cases. For example it is raised in our internal conformance checks. Not sure on best alternative here.